### PR TITLE
Feature/pro 1480/sd burn

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

**Comment by elffjs** [comment]
Conflict!

**Comment by elffjs** [comment]
Oh it's just git being dumb I think.

File: contracts/implementations/nodes/SyntheticDevice.sol [hunk 1]
```diff
@@ -82,6 +91,79 @@ contract SyntheticDevice is AccessControlInternal {
...
+     * @notice Mints a list of synthetic devices and links them to vehicles
+     * To be called for existing vehicles already connected
+     * @dev Caller must have the admin role
+     * @dev All devices will be minted under the same integration node
+     * @param data input data with the following fields:
+     *  vehicleNode -> Vehicle node id
+     *  syntheticDeviceAddr -> Address associated with the synthetic device
+     *  attrInfoPairs -> List of attribute-info pairs to be added
+     */
+    function mintSyntheticDeviceBatch(
```
Comment (elffjs) [inline contracts/implementations/nodes/SyntheticDevice.sol:104]: Hmm let's merge in dev. Is this `@param` still complete?

File: contracts/implementations/nodes/SyntheticDevice.sol [hunk 2]
```diff
@@ -166,6 +248,62 @@ contract SyntheticDevice is AccessControlInternal {
...
+            .idProxyAddress;
+        address sdIdProxyAddress = sds.idProxyAddress;
+
+        if (
+            ms.nodeLinks[vehicleIdProxyAddress][sdIdProxyAddress][
+                vehicleNode
+            ] != syntheticDeviceNode
+        ) revert Errors.VehicleNotPaired(vehicleNode);
+
+        address owner = INFT(sdIdProxyAddress).ownerOf(syntheticDeviceNode);
```
Comment (elffjs) [inline contracts/implementations/nodes/SyntheticDevice.sol:280]: Note that this is the first place where we would catch `vehicleNode = syntheticDeviceNode = 0` or simply `syntheticDeviceNode = 0` and the vehicle being unpaired; this line exploding.

File: contracts/implementations/nodes/SyntheticDevice.sol [hunk 3]
```diff
@@ -166,6 +248,62 @@ contract SyntheticDevice is AccessControlInternal {
...
+        bytes32 message = keccak256(
+            abi.encode(BURN_TYPEHASH, vehicleNode, syntheticDeviceNode)
+        );
+
+        if (!Eip712CheckerInternal._verifySignature(owner, message, ownerSig))
+            revert Errors.InvalidOwnerSignature();
+
+        ns.nodes[sdIdProxyAddress][syntheticDeviceNode].parentNode = 0;
```
Comment (elffjs) [inline contracts/implementations/nodes/SyntheticDevice.sol:288]: Do we get gas back from this? It's a lot of work.

File: contracts/implementations/nodes/SyntheticDevice.sol (hunk 1 above)
Comment (LorranSutter) [reply contracts/implementations/nodes/SyntheticDevice.sol:104]: Ops

File: contracts/implementations/nodes/SyntheticDevice.sol (hunk 2 above)
Comment (LorranSutter) [reply contracts/implementations/nodes/SyntheticDevice.sol:280]: That's a catch

**Review by elffjs — APPROVED** [review]

**Pull request merged**